rotkehlchen/exchanges/binance.py

```diff
@@ -589,6 +607,184 @@ def _query_lending_balances(
 
         return balances
 
+    def _api_query_list_paginated_within_time_delta(
+            self,
+            start_ts: Timestamp,
+            end_ts: Timestamp,
+            time_delta: int,
+            api_type: BINANCE_API_TYPE,
+            method: Literal['lending/union/interestHistory'],
+            additional_options: Optional[dict] = None,
+    ) -> list[dict[str, Any]]:
+        """Request via `_api_query_list_within_time_delta` using `current` param for pagination"""
+        data = []
+        current_page = 1
+        while True:
+            if additional_options:
+                additional_options.update({'current': current_page})
+
+            response = self._api_query_list_within_time_delta(
+                api_type=api_type,
+                start_ts=start_ts,
+                end_ts=end_ts,
+                time_delta=time_delta,
+                method=method,
+                additional_options=additional_options,
+            )
+            # we get an empty response, and `current` is at least 1
+            if len(response) == 0 and current_page >= 1:
```

Comment:
Isn't it always true that `current_page >= 1`? I believe you can remove this condition

```suggestion
            if len(response) == 0:
```

either you wanted to set current page to 0 or this condition is wrong. Also what happens when you don't have any events? does this for loop stop (I mean if you change the condition)?

Reply:
Yeah, you're right. The `and` case isn't necessary here